### PR TITLE
Initial packaging

### PR DESCRIPTION
**Pull request by Mailaender**
https://github.com/eclipse-theia/theia/issues/13862

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: snapcore/action-build@v1
+      id: build
+    - uses: snapcore/action-publish@v1
+      env:
+        SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
```
Comment (jfaltermeier) [inline .github/workflows/publish.yml:20]: Do you know if this secret has already been added to the repository?

**Review by jfaltermeier — APPROVED** [review]
As far as I can tell, this looks good.
I tried testing it on my 22.04 machine, but I keep getting "Timed out waiting for networking to be ready" after running `snapcraft` at the moment.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (Mailaender) [reply .github/workflows/publish.yml:20]: I requested it at https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/5021#note_2825046

**Comment by Mailaender** [comment]
@jfaltermeier I assume you may be running Docker as well. There is a known issue https://documentation.ubuntu.com/lxd/en/latest/howto/network_bridge_firewalld/#prevent-connectivity-issues-with-lxd-and-docker

File: .github/workflows/publish.yml (hunk 1 above)
Comment (Mailaender) [reply .github/workflows/publish.yml:20]: It has been setup https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/5021#note_2828682 we can only test it by merging this.

**Comment by Mailaender** [comment]
I am not acquainted with that error. Did you install as instructed?
> sudo snap install theia-ide*.snap --dangerous --classic

**Comment by jfaltermeier** [comment]
Yes, but I did get a warning:

```
$ sudo snap install theia-ide*.snap --dangerous --classic
Warning: flag --classic ignored for strictly confined snap theia-ide
```

The .deb works fine.

When I changed the `confinement: strict` to `classic` in the `snapcraft.yaml`, it worked for me. Based on what I’ve read about the differences between `strict` and `classic` confinement, it seems like `classic` makes sense for an IDE, right? Since it needs access to source code files and possibly tools for compiling.

**Comment by Mailaender** [comment]
Updated this for [classic confinement](https://snapcraft.io/docs/classic-confinement) as it was just said to be approved at https://forum.snapcraft.io/t/classic-confinement-for-eclipse-theia-ide/42177/6

**Review by jfaltermeier — APPROVED** [review]
LGTM

**Pull request merged**